test/csuite/wt4803_cache_overflow_abort/main.c

```diff
@@ -0,0 +1,177 @@
+#include "test_util.h"
+
+#include <signal.h>
+#include <sys/wait.h>
+
+#define	NUM_KEYS	2000
+
+static int
+handle_message(WT_EVENT_HANDLER *handler, WT_SESSION *session, int error,
+    const char *message)
+{
+	WT_UNUSED(handler);
+	WT_UNUSED(session);
+
+	(void)fprintf(
+	    stderr, "%s: %s\n", message, session->strerror(session, error));
+
+	if (error == WT_PANIC &&
+	    strstr(message, "exceeds maximum size") != NULL) {
+		fprintf(
+		    stderr, "Got cache overflow error. Exiting with EINVAL.\n");
+		exit(EINVAL);
+	}
+
+	return (0);
+}
+
+static WT_EVENT_HANDLER event_handler = {
+	handle_message,
+	NULL,
+	NULL,
+	NULL
+};
+
+static void
+las_workload(TEST_OPTS *opts, const char *las_file_max)
+{
+	WT_CURSOR *cursor;
+	WT_SESSION *other_session, *session;
+	int i;
+	char buf[WT_MEGABYTE], open_config[128];
+
+	testutil_check(__wt_snprintf(open_config, sizeof(open_config),
+	    "create,cache_size=50MB,cache_overflow=(file_max=%s)",
+	    las_file_max));
+
+	testutil_check(wiredtiger_open(
+	    opts->home, &event_handler, open_config, &opts->conn));
+	testutil_check(
+	    opts->conn->open_session(opts->conn, NULL, NULL, &session));
+	testutil_check(
+	    session->create(session, opts->uri, "key_format=i,value_format=S"));
+	testutil_check(
+	    session->open_cursor(session, opts->uri, NULL, NULL, &cursor));
+
+	memset(buf, 0xA, WT_MEGABYTE);
+	buf[WT_MEGABYTE - 1] = '\0';
+
+	/* Populate the table. */
+	for (i = 0; i < NUM_KEYS; ++i) {
+		cursor->set_key(cursor, i);
+		cursor->set_value(cursor, buf);
+		testutil_check(cursor->insert(cursor));
+	}
+
+	/*
+	 * Open a snapshot isolation transaction in another session. This forces
+	 * the cache to retain all previous values. Then update all keys with a
+	 * new value in the original session while keeping that snapshot
+	 * transaction open. With the large value buffer, small cache and lots
+	 * of keys, this will force a lot of lookaside usage.
+	 *
+	 * When the file_max setting is small, the maximum size should easily be
+	 * reached and we should panic. When the maximum size is large or not
+	 * set, then we should succeed.
+	 */
+	testutil_check(
+	    opts->conn->open_session(opts->conn, NULL, NULL, &other_session));
+	other_session->begin_transaction(other_session, "isolation=snapshot");
```

Comment:
This should have a `testutil_check` wrapper.

Reply:
98d8d9d